### PR TITLE
WAG-349 : Add redirects for old rules PDF URLs

### PR DESCRIPTION
**Pull request by btejha**
**Key Updates:**

- Eliminated CSV Parsing: The CloudFront function now directly contains the redirect logic; no external CSV is parsed at runtime.
- Canonical URL Redirects (HTTP 301):
      •	Legacy rule URLs (e.g., /files/documents/Rule-13_Amended_08082024.pdf) now redirect to clean versions (e.g., /documents/rule-13.pdf).
      • All 301 redirect targets consistently start with /documents/, removing the /files/ prefix from the final URL shown to the user.

- The script:
     Outputs:
	•	pdf_redirect_function.js: JavaScript to be deployed as a CloudFront Function
	•	pdf_redirects.json: backing map of redirect paths

- URL Normalization:
       •	Handles case insensitivity (e.g., "Rule" to "rule"). 
       •	Strips various suffixes (e.g., _Amended, superseded, New) using a robust regex. 
       •	Standardizes formatting (e.g., Rule-1-1.pdf becomes rule-1.1.pdf).
 - Hardcoded redirect entries remain supported for non-patterned edge cases
       •	Applied consistent transformation logic (e.g., Rule-1_Amended_03202023.pdf → rule-1.pdf)
       • 	Capital letter normalization and suffix cleanup (e.g., _Amended, 2nd-Amended, New) handled via regex
       •	Unit tests added to cover regex redirect behavior and edge cases

- Unit Tests: Expanded tests verify both patterned and hardcoded redirects, ensuring correct /documents/ target paths for all 301 responses

### CODE REVIEW

File: website/redirects/pdf_redirect_function.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+              statusCode: 302,
+              statusDescription: "Found",
+              headers: {
+                location: { value: "/files" + newUri }
+              }
+            };
+          }
+
+          if (genericPattern.test(uri)) {
+            var newUri = uri.replace("/documents/Rule-", "/documents/rule-"); // lowercase only "Rule"
```
Comment (guruparan18) [inline website/redirects/pdf_redirect_function.js:49]: OK, I see this. So `/Rule-140-new.pdf` --> `/rule-140-new.pdf` --> `/rule-140.pdf`? Can this be merged together.

File: website/home/management/commands/create_rules_pdf_redirects.py
[the remark this answers is not in the document]
Comment (btejha) [reply website/home/management/commands/create_rules_pdf_redirects.py:1]: Moved to utils

**Review by guruparan18 — APPROVED** [review]
Looks good. Thanks.

**Review by jtdevos — APPROVED** [review]
